src/Centreon/Domain/Repository/ServiceRepository.php

```diff
@@ -162,4 +145,95 @@ public function getChainByParant($id, &$result)
 
         return $result;
     }
-}
+
+    /**
+     * Get a chain of the related objects
+     *
+     * @param int[] $pollerIds
+     * @param int[] $ba
+     * @return array
+     */
+    public function getServiceIdsByPoller(array $pollerIds, array $ba = null): array
+    {
+        // prevent SQL exception
+        if (!$pollerIds) {
+            return [];
+        }
+
+        $ids = implode(',', $pollerIds);
+        $sql = <<<SQL
+SELECT l.* FROM (
+SELECT DISTINCT t.service_id AS `id`
+FROM service AS t
+INNER JOIN host_service_relation AS hsr ON hsr.service_service_id = t.service_id
+INNER JOIN ns_host_relation AS hr ON hr.host_host_id = hsr.host_host_id
+WHERE hr.nagios_server_id IN ({$ids})
+SQL;
+
+        // Extract BA services
```

Comment:
can you find a way to remove references to centreon-bam ?

Reply:
Could be impossible with actual system without rewrite some part.
May be for latter